### PR TITLE
Misc battle and battle palace documentation

### PR DESCRIPTION
**Pull request by GriffinRichards**
- Clean up some data in `battle_script_commands.c`
- Convert battle constants from hex values to bit shifts
- Add new battle constants for multi-turn statuses
- Added a few other misc constants / constant usage
- Document how battle palace chooses moves

### CODE REVIEW

File: include/constants/battle.h [hunk 1]
```diff
@@ -37,146 +37,156 @@
...
+#define B_OUTCOME_LOST                 2
+#define B_OUTCOME_DREW                 3
+#define B_OUTCOME_RAN                  4
+#define B_OUTCOME_PLAYER_TELEPORTED    5
+#define B_OUTCOME_MON_FLED             6
+#define B_OUTCOME_CAUGHT               7
+#define B_OUTCOME_NO_SAFARI_BALLS      8
+#define B_OUTCOME_FORFEITED            9
+#define B_OUTCOME_MON_TELEPORTED       10
+#define B_OUTCOME_LINK_BATTLE_RAN      128
```
Comment (DizzyEggg) [inline include/constants/battle.h:93]: Makes more sense as 0x80 or (1 << 7) as it's used as the last bit.

File: include/constants/battle.h [hunk 2]
```diff
@@ -215,71 +225,70 @@
...
+#define MOVE_EFFECT_THRASH              53
+#define MOVE_EFFECT_KNOCK_OFF           54
+#define MOVE_EFFECT_NOTHING_37          55
+#define MOVE_EFFECT_NOTHING_38          56
+#define MOVE_EFFECT_NOTHING_39          57
+#define MOVE_EFFECT_NOTHING_3A          58
+#define MOVE_EFFECT_SP_ATK_TWO_DOWN     59
+#define NUM_MOVE_EFFECTS                60
+
+#define MOVE_EFFECT_AFFECTS_USER        64
```
Comment (DizzyEggg) [inline include/constants/battle.h:290]: Again, used as a flag

File: include/constants/battle.h [hunk 3]
```diff
@@ -215,71 +225,70 @@
...
+#define MOVE_EFFECT_CERTAIN             128
```
Comment (DizzyEggg) [inline include/constants/battle.h:291]: flag as well

File: include/constants/pokemon.h [hunk 4]
```diff
@@ -80,7 +81,7 @@
 #define STAT_ACC     6 // Only in battles.
 #define STAT_EVASION 7 // Only in battles.
 
-#define NUM_EV_STATS     NUM_STATS - 1 // excludes HP
+#define NUM_NATURE_STATS NUM_STATS - 1 // excludes HP
```
Comment (DizzyEggg) [inline include/constants/pokemon.h:84]: hmmm? why nature instead of ev?

Comment (GriffinRichards) [reply include/constants/pokemon.h:84]: HP has an EV, but not a Nature modification. It's named incorrectly.

**Pull request merged**